src/app/api/users/users.service.ts

```diff
@@ -33,14 +32,31 @@ class UsersService extends BaseService {
 
     // Filter out companies where isPlaceholder is true if companies.data is not null
     const filteredCompanies = companies.data ? companies.data.filter((company) => !company.isPlaceholder) : []
-    const clientsWithCompanyData = clients.data?.map((client) => {
-      const companyForClient = filteredCompanies.find((company) => company.id === client.companyId)
-      return { ...client, fallbackColor: companyForClient?.fallbackColor }
-    })
+    const clientsWithCompanyData =
+      clients.data?.map((client) => {
+        const companyForClient = filteredCompanies.find((company) => company.id === client.companyId)
+        return { ...client, fallbackColor: companyForClient?.fallbackColor }
+      }) || []
 
     return { internalUsers: ius.data, clients: clientsWithCompanyData, companies: filteredCompanies }
   }
 
+  /**
+   * Service to filter users starting with a given keyword
+   * @param keyword
+   * @returns
+   */
+  async getFilteredUsersStartingWith(keyword: string) {
```

Comment:
this returns IU, CU, and companies, so calling the func `getFilteredUsersStartingWith` was a little unclear to me at first, and only after jumping up and down in the code did I realize what this was doing.